src/core/egg-herbie.rkt

```diff
@@ -43,35 +32,16 @@
 ;; - FFIRule: struct defined in egg-herbie
 ;; - EgraphIter: struct defined in egg-herbie
 
-(define (make-raw-string s)
-  (define b (string->bytes/utf-8 s))
-  (define n (bytes-length b))
-  (define ptr (malloc 'raw (+ n 1)))
-  (memcpy ptr b n)
-  (ptr-set! ptr _byte n 0)
-  ptr)
-
-(define (make-ffi-rule rule)
-  (define name (make-raw-string (~a (rule-name rule))))
-  (define lhs (make-raw-string (~a (rule-input rule))))
-  (define rhs (make-raw-string (~a (rule-output rule))))
-  (define p (make-FFIRule name lhs rhs))
-  (register-finalizer p free-ffi-rule)
-  p)
-
-(define (free-ffi-rule rule)
-  (free (FFIRule-name rule))
-  (free (FFIRule-left rule))
-  (free (FFIRule-right rule))
-  (free rule))
-
 ;; Wrapper around Rust-allocated egg runner
 (struct egraph-data
         (egraph-pointer ; FFI pointer to runner
          herbie->egg-dict ; map from symbols to canonicalized names
          egg->herbie-dict ; inverse map
          id->spec)) ; map from e-class id to an approx-spec or #f
 
+;; Racket representation of per-iteration runner data
+(struct iteration-data (num-nodes num-eclasses time))
+
```

Comment:
Ok, at this point we should move this into `egg-herbie` and return this instead of messing with hash tables.

Reply:
Ok let me do that